src/commands.rs

```diff
@@ -4150,6 +4151,108 @@ fn do_git_clone(
     Ok((workspace_command, maybe_default_branch))
 }
 
+// Wrapper around git::fetch that adds progress feedback on TTYs
+fn git_fetch(
+    mut_repo: &mut MutableRepo,
+    git_repo: &git2::Repository,
+    remote_name: &str,
+) -> Result<Option<String>, GitFetchError> {
+    let mut callback = None;
+    let mut printed = false;
+    if atty::is(atty::Stream::Stdout) {
+        let mut rate = RateEstimate::new();
+        let printed = &mut printed;
+        callback = Some(move |progress: &git::Progress| {
+            if let Some(estimate) = rate.update(Instant::now(), progress.bytes_downloaded) {
+                let (scaled, prefix) = binary_prefix(estimate);
+                const HIDE_CURSOR: &str = "\x1b[25l";
+                const MOVE_TO_START: &str = "\x1b[G";
+                const SHOW_CURSOR: &str = "\x1b[25h";
+                const CLEAR_TRAILING: &str = "\x1b[K";
+                print!(
+                    "{}{}{: >3.0}% at {: >5.1} {}B/s{}{}",
+                    HIDE_CURSOR,
+                    MOVE_TO_START,
+                    100.0 * progress.overall_progress,
+                    scaled,
+                    prefix,
+                    CLEAR_TRAILING,
+                    SHOW_CURSOR,
+                );
```

Comment:
Despite my efforts, this yields a flickery cursor, at least in termite. Better ideas from ANSI wizards welcome, though I don't think this is a blocker. We could hide the cursor for the duration if necessary, but then we need to take measures to ensure panics, ^C, etc. don't leave the user's terminal in a bad state.

Reply:
Judging this Good Enough for the first pass.